File: 0870-Advantage Shuffle/0870-Advantage Shuffle.cpp
class Solution {
public:
    vector<int> advantageCount(vector<int>& A, vector<int>& B) {
        int n = A.size();
        sort(A.begin(), A.end());
        priority_queue<pair<int, int>> pq;
        for (int i = 0; i < n; ++i) {
            pq.emplace(B[i], i);
        }
        vector<int> result(n);
        int lo = 0, hi = n - 1;
        while (!pq.empty()) {
            int val = pq.top().first;
            int index = pq.top().second;
            pq.pop();
            if (A[hi] > val) {
                result[index] = A[hi--];
            }
            else {
                result[index] = A[lo++];
            }
        }
        return result;
    }
};

File: 0801-0900/0870-Advantage Shuffle/0870-Advantage Shuffle.py
import heapq
class Solution:
    def advantageCount(self, A: List[int], B: List[int]) -> List[int]:
        A.sort()
        pq = [(-b, i) for i, b in enumerate(B)]
        heapq.heapify(pq)
        low = 0
        high = len(A) - 1
        result = [0] * len(A)
        while pq:
            b, i = heapq.heappop(pq)
            b = -b
            if A[high] > b:
                result[i] = A[high]
                high -= 1
            else:
                result[i] = A[low]
                low += 1
        return result
